Gems/Atom/RPI/Code/Source/RPI.Edit/Material/MaterialSourceData.cpp

```diff
@@ -77,6 +74,55 @@ namespace AZ
             }
         }
         
+        void MaterialSourceData::SetPropertyValue(const Name& propertyId, const MaterialPropertyValue& value)
+        {
+            if (!propertyId.IsEmpty())
+            {
+                m_propertyValues[propertyId] = value;
+            }
+        }
+
+        const MaterialPropertyValue& MaterialSourceData::GetPropertyValue(const Name& propertyId) const
+        {
+            auto iter = m_propertyValues.find(propertyId);
+            if (iter == m_propertyValues.end())
+            {
+                return m_invalidValue;
```

Comment:
Out of curiosity, why is a sentinel value for "not found" used instead of returning a `const MaterialPropertyValue*`?

Reply:
Partly just out of habit, as we do this pattern a lot in Atom. Also because MaterialPropertyValue is a type that has a natural invalid state, so if we return MaterialPropertyValue* then client code has to do both a null check and a check for validity of the MaterialPropertyValue. We have lots of APIs that take MaterialPropertyValue& not MaterialPropertyValue*, and those APIs already check for MaterialPropertyValue validity.